### PR TITLE
dataflow-types: Protobuf support for ComputeCommand (w/ proptest).

### LINKED ISSUE #11737: Add protobuf support for `mz_dataflow_types::client::ComputeCommand`
Add protobuf support for [`mz_dataflow_types::client::ComputeCommand`](https://github.com/materializeinc/materialize/blob/d0749fc45496de0601370b9f482eb5c354dc3fa7/src/dataflow-types/src/client.rs#L108-L141).

Blocked by:
- #11971

### PR DESCRIPTION
**Pull request by wangandi**
This PR adds a known-desirable feature.
 
Closes #11737. 

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

No user-facing behavior changes

### CODE REVIEW

File: src/dataflow-types/src/client.rs [hunk 1]
```diff
@@ -176,6 +176,125 @@ pub enum ComputeCommand<T = mz_repr::Timestamp> {
...
+    fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+        prop_oneof![
+            any::<Option<LoggingConfig>>()
+                .prop_map(|logging| ComputeCommand::CreateInstance(logging)),
+            Just(ComputeCommand::DropInstance),
+            proptest::collection::vec(
+                any::<DataflowDescription<Plan, CollectionMetadata, mz_repr::Timestamp>>(),
+                1..4
+            )
+            .prop_map(|dataflows| ComputeCommand::CreateDataflows(dataflows)),
```
Comment (aalexandrov) [inline src/dataflow-types/src/client.rs:273]: [Eta-reduce](https://en.wikipedia.org/wiki/Lambda_calculus#%CE%B7-reduction) this (required by the linter): 

```rust
prop_map(ComputeCommand::CreateDataflows)
```

File: src/dataflow-types/src/client.rs [hunk 2]
```diff
@@ -176,6 +176,125 @@ pub enum ComputeCommand<T = mz_repr::Timestamp> {
...
+}
+
+impl Arbitrary for ComputeCommand<mz_repr::Timestamp> {
+    type Strategy = BoxedStrategy<Self>;
+    type Parameters = ();
+
+    fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+        prop_oneof![
+            any::<Option<LoggingConfig>>()
+                .prop_map(|logging| ComputeCommand::CreateInstance(logging)),
```
Comment (aalexandrov) [inline src/dataflow-types/src/client.rs:267]: [Eta-reduce](https://en.wikipedia.org/wiki/Lambda_calculus#%CE%B7-reduction) this (required by the linter): 

```rust
prop_map(ComputeCommand::CreateInstance)
```

File: src/dataflow-types/src/client.rs [hunk 3]
```diff
@@ -176,6 +176,125 @@ pub enum ComputeCommand<T = mz_repr::Timestamp> {
...
+                ),
+                1..4
+            )
+            .prop_map(|collections| ComputeCommand::AllowCompaction(
+                collections
+                    .into_iter()
+                    .map(|(id, frontier_vec)| { (id, Antichain::from(frontier_vec)) })
+                    .collect()
+            )),
+            any::<Peek>().prop_map(|peek| ComputeCommand::Peek(peek)),
```
Comment (aalexandrov) [inline src/dataflow-types/src/client.rs:287]: [Eta-reduce](https://en.wikipedia.org/wiki/Lambda_calculus#%CE%B7-reduction) this (required by the linter): 

```rust
prop_map(ComputeCommand::Peek)
```

**Review by aalexandrov — APPROVED** [review]
A couple of minor things, otherwise LGTM - I can't believe we are (almost) through with this! 

:champagne: :clinking_glasses: :champagne:

File: src/dataflow-types/src/client.rs [hunk 4]
```diff
@@ -176,6 +176,125 @@ pub enum ComputeCommand<T = mz_repr::Timestamp> {
...
+                        .iter()
+                        .map(|(id, frontier)| ProtoCompaction {
+                            id: Some(id.into()),
+                            frontier: Some(frontier.into()),
+                        })
+                        .collect(),
+                }),
+                ComputeCommand::Peek(peek) => Peek(peek.into()),
+                ComputeCommand::CancelPeeks { uuids } => CancelPeeks(ProtoCancelPeeks {
+                    uuids: uuids.into_iter().map(|id| id.into_proto()).collect(),
```
[the remark this answers is not in the document]
Comment (wangandi) [reply src/dataflow-types/src/client.rs:205]: Using `map(ProtoRepr::into_proto)` results in a compile error that looks like this:
```
error[E0277]: the trait bound `&usize: ProtoRepr` is not satisfied
   --> src/dataflow-types/src/plan/top_k.rs:147:47
    |
147 |             group_key: x.group_key.iter().map(ProtoRepr::into_proto).collect(),
    |                                               ^^^^^^^^^^^^^^^^^^^^^ the trait `ProtoRepr` is not implemented for `&usize`
    |
    = help: the following implementations were found:
              <usize as ProtoRepr>
              <u128 as ProtoRepr>
              <u8 as ProtoRepr>
```

**Pull request merged**